processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionContext.java

```diff
@@ -157,6 +160,13 @@ public void resume() {
         metrics.partitionPausedTime.record(pausedNanos, TimeUnit.NANOSECONDS);
     }
 
+    public void reloading(boolean reloading) {
+        this.reloading = reloading;
+        if (reloading) {
+            pause();
```

Comment:
pausing here by PartitionContext itself doesn't work to pause fetching actually.

Decaton's fetch-pause works like:
- (1) ConsumeManager retrieves partition list that needs pause https://github.com/line/decaton/blob/v6.2.0/processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ConsumeManager.java#L182
- (2) Call Consumer#pause to stop fetching further records. From now on, Consumer#poll doesn't return records for these partitions

So, if we pause here by PartitionContext itself, it will not be included in (1) (because it's already "paused").

Correct approach here may be holding per-partition `reloading` flag in PartitionContexts, and include them in `partitionsNeedsPause`